backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_OrderLine.java

```diff
@@ -156,19 +157,22 @@ public void checkHaddexDate(final I_C_OrderLine orderLine)
 
 		if (product.isHaddexCheck() && product.getDateHaddexCheck() != null)
 		{
-			final long hadexAllowedDifference = Math.abs(ChronoUnit.MONTHS.between(SystemTime.asZonedDateTime(),
-					TimeUtil.asZonedDateTime(product.getDateHaddexCheck())));
-			if (hadexAllowedDifference > getMaxHaddexAge(orderLine.getAD_Client_ID(), orderLine.getAD_Org_ID()))
+			final long differenceBetweenNowAndPromisedDateInMonths = Math.abs(
+					ChronoUnit.MONTHS.between(
+							SystemTime.asZonedDateTime(),
+							TimeUtil.asZonedDateTime(orderLine.getDatePromised())
+							));
```

Comment:
why is the product.getDateHaddexCheck() no longer used?

Reply:
Tobi suggested and Mark confirmed.
https://github.com/metasfresh/me03/issues/6008#issuecomment-743019473